pkg/destroy/gcp/address.go

```diff
@@ -72,13 +72,14 @@ func (o *ClusterUninstaller) destroyAddresses() error {
 		return err
 	}
 	items := o.insertPendingItems("address", found)
-	errs := []error{}
 	for _, item := range items {
 		err := o.deleteAddress(item)
 		if err != nil {
-			errs = append(errs, err)
+			o.errorTracker.suppressWarning(item.key, err, o.Logger)
```

Comment:
why do we have to suppress the warning in each handler instead of handling it at the caller sites?

Reply:
We don't know the "item.key" at the caller sites... so we have no way of knowing what object the error is related to. We can't use the error itself (as we found in AWS), because the error messages are sometimes dynamic and some errors will never bubble up. Thus why we have to know which object the error is related to. As a result, we aren't really bubbling up a specific error. Instead, we are bubbling up "this object has seen errors".

Reply:
What would it take to bubble up right error messages so that we can suppress the errors if we have to at higher level instead of inside each handler

Reply:
We could refactor such that we have a single list of items to delete by `SelfLink` which includes `type`, and loop through those with a switch based on type that calls the appropriate delete function. The delete function could then just return an error and the main loop could handle the suppression. We would have to expand our discovery phase to populate that list. That sounds like a lot of work.

Reply:
why would we have to include a new discovery.. the delete knows which items was being deleted.. would adding context about the item being deleted when returning error not be enough..

Reply:
The way it was before, we were returning aggregated errors as a single error. Breaking that down into individual components would require significant work as well.

Reply:
Just as a way to think through, how would we go about changing the code so that we didn't have to suppress individually and can actually do this handling at higher level outside these runnners? So can we can shape follow-up to improve it.